### ISSUE TITLE
How to setup this addon

### ISSUE THREAD

**Issue by ghost** [ZappBrannigannnn/SportsView#1]
I am trying to use your app but I am not sure how to setup. I started the addon and I got a python error with textsize and I saw this has been changed to textlength so I made the changes in your files. When I start the addon, I just get a blank screen with all sports, my matches, and favorites with a hoops image in the background. I am pointing the SPORTS folder to your demo SPORTS folder. Do I need an API key or can I use what you have as default?

**Comment by ZappBrannigannnn** [comment]
No the addon has its own dedicated api key. On the first run the addon can
take a minute or 2 to download all the images. I'm not sure what
"textlength" your referring to. Did you download the latest version today?
Are there any errors in the log? Enable debug logging restart kodi. Start
the addon on look for an error in the log. Then paste here I guess. I'm not
an expert at making add-ons or Github. This is the first message I've got
on here hopefully this method of communication works for log snippets.

**Comment by ZappBrannigannnn** [comment]
It's strange that the landing page has a basketball background. The landing
page should be grass.

**Comment by ghost** [comment]
This is the python error I received in the kodi log

 'ImageDraw' object has no attribute 'textsize' in mymatches/mymatchesseasons.py 

I did a search and it said that textsize is deprecated and to use textlength so I changed them.

I stand corrected. I get the grassy field and then when I click on all sports. I get redirected to a page with a basketball hoop in the background. The images never load and I am not seeing any errors in the kodi.log as I have tuned on debug.

**Comment by ZappBrannigannnn** [comment]
Maybe wait longer especially for "all sports" it has a lot to download from
the website and at the moment is just a proof of concept that I'm slowly
implementing. I right I'd focus on "my matches"

**Comment by ghost** [comment]
I let it run for 30 minutes and nothing happened.

I tried going to my matches but it gives me a python error.

`error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'IndexError'>
                                                   Error Contents: list index out of range
                                                   Traceback (most recent call last):
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 44, in onAction
                                                       self.sports_buttons.moveFocus(1, 0)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 231, in moveFocus
                                                       self.parent_window.setFocusId(self.buttons[self.focused_index].getId())
                                                                                     ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
                                                   IndexError: list index out of range
                                                   -->End of Python script error report<--
`

**Comment by ZappBrannigannnn** [comment]
I think I remember getting this error when the folder system was not set
correctly in the settings. Maybe try to set the folders again. Remember the
SPORTS folder should be what you're pointing to not the folders in it...
Did the all sports page work at all?

**Comment by ZappBrannigannnn** [comment]
This error is a movefocus error. Likely meaning it's trying to move the
focus between items that don't exist cause they weren't populated properly.
This points to the folder setting issue again. Don't give up! Haha

**Comment by ghost** [comment]
No the sports all didn't work.

Where does the SPORTS folder setting need to point to. I have tried /home/path/.kodi/addons/plugin.sportsview/SPORTS and also I have tried /home/path/.kodi/addons/plugin.sportsview/SPORTS/MLB

**Comment by ZappBrannigannnn** [comment]
What machine are you using?

It should be for example:

C:/SPORTS/

inside it should be sports like baseball, rugby, ice hockey, motorsport,
soccer.

Not leagues.

Attached is a photo of what the all sports page should look like.

**Comment by ZappBrannigannnn** [comment]
If you want to test the issue with the all sports page you'll have to
enable debug logging, close kodi, repoen kodi, open the sportsview addon,
go to all sports and wait 2 minutes. Then look in the log and send me the
error if there is one.

**Comment by ghost** [comment]
I'm running Arch Linux 64 bit on a Raspberry Pi so ARM. I am also using Python 3.11.8

I can't see your attachment.

**Comment by ZappBrannigannnn** [comment]
![20240323_224254](https://github.com/ZappBrannigannnn/SportsView/assets/136344138/50022e20-3589-4727-82d3-28f44c36ff34)

**Comment by ZappBrannigannnn** [comment]
> I'm running Arch Linux 64 bit on a Raspberry Pi so ARM. I am also using Python 3.11.8
> 
> I can't see your attachment.

I see. That I have not tested it on. It works on Nvidia shield (android), Windows 10, and Windows 11. I would have thought that once in the kodi environment it would be good to go. Hopefully that's the case. I'm interested to see the all sports errors tbh.

**Comment by ghost** [comment]
I think this could be a problem in linux

custom_font_path = xbmcvfs.translatePath("special://home/addons/plugin.sportsview/resources/fonts/ariblk.ttf")

If I have it correctly, the special: points to /home/user/.kodi

and the above line would point to 

/home/user/.kodi/home/addons/plugin.sportsview/resources/fonts/ariblk.ttf which doesn't exist. I will test tomorrow.

**Comment by ZappBrannigannnn** [comment]
Maybe... But, I don't think that should be a problem according to this: https://kodi.wiki/view/Special_protocol ... the special protocol should work regardless of OS. 

Did you get an error regarding the custom font directory?

In your ...\.kodi\addons\plugin.sportsview\resources\fonts\    folder is the font there??

**Comment by ZappBrannigannnn** [comment]
"home" should just differentiate between the few access points that "special" points to.... I THINK!! haha

**Comment by ghost** [comment]
> Maybe... But, I don't think that should be a problem according to this: https://kodi.wiki/view/Special_protocol ... the special protocol should work regardless of OS.
> 
> Did you get an error regarding the custom font directory?
> 
> In your ....kodi\addons\plugin.sportsview\resources\fonts\ folder is the font there??

Yeah I think you are correct. I will try again in the morning and check the debug log file to see if it throws an error. It never gave me a missing font file error.

**Comment by ZappBrannigannnn** [comment]
okay. cool. you had me worried. cause getting kodi to use custom font was a major pain. But i was being stubborn about it.

let me know how it goes!

**Comment by ghost** [comment]
These are the only entries I see when I select All Sports

`2024-03-23 23:32:19.463 T:405     debug <general>: Activating window ID: 13030
2024-03-23 23:32:19.463 T:405     debug <general>: ------ Window Deinit (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/landingpag>
2024-03-23 23:32:19.463 T:405     debug <general>: ------ Window Init (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.xm>
2024-03-23 23:32:19.463 T:405      info <general>: Loading skin file: /home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.xml>
2024-03-23 23:32:19.463 T:343439   debug <general>: AllSportsWindow - onInit
2024-03-23 23:32:19.463 T:343439   debug <general>:

2024-03-23 23:32:19.879 T:343432   debug <CWebserver[8080]>: request received for /jsonrpc
2024-03-23 23:32:20.257 T:343439   debug <general>: ALLSPORTS_LIST.TXT AND SPORTS_LIST MATCH
2024-03-23 23:32:20.257 T:343439   debug <general>:

2024-03-23 23:32:20.264 T:343439   debug <general>: ffmpeg[0x555622be1b98]: [swscaler] No accelerated colorspace conversion found from yuv420p to bgra.


2024-03-23 23:32:13.663 T:343432   debug <CWebserver[8080]>: request received for /jsonrpc
2024-03-23 23:32:13.669 T:405     debug <general>: Activating window ID: 10000
2024-03-23 23:32:13.669 T:405     debug <general>: ------ Window Deinit (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.>
2024-03-23 23:32:13.669 T:405     debug <general>: ------ Window Init (Home.xml) ------
2024-03-23 23:32:13.750 T:343433   debug <general>: Thread JobWorker start, auto delete: true
2024-03-23 23:32:13.750 T:343433   debug <general>: [threads] name: 'JobWorker' priority: '1'
2024-03-23 23:32:13.760 T:343433   debug <general>: ffmpeg[0x555620f71a30]: [swscaler] No accelerated colorspace conversion found from yuv420p to bgra.
2024-03-23 23:32:13.881 T:343432   debug <CWebserver[8080]>: request received for /jsonrpc
2024-03-23 23:32:13.663 T:343432   debug <CWebserver[8080]>: request received for /jsonrpc
2024-03-23 23:32:13.669 T:405     debug <general>: Activating window ID: 10000
2024-03-23 23:32:13.669 T:405     debug <general>: ------ Window Deinit (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.>
2024-03-23 23:32:13.669 T:405     debug <general>: ------ Window Init (Home.xml) ------
2024-03-23 23:32:13.750 T:343433   debug <general>: Thread JobWorker start, auto delete: true
2024-03-23 23:32:13.750 T:343433   debug <general>: [threads] name: 'JobWorker' priority: '1'
2024-03-23 23:32:13.760 T:343433   debug <general>: ffmpeg[0x555620f71a30]: [swscaler] No accelerated colorspace conversion found from yuv420p to bgra.
2024-03-23 23:32:13.881 T:343432   debug <CWebserver[8080]>: request received for /jsonrpc

2024-03-23 23:46:38.201 T:344540    info <general>: Skipped 3 duplicate messages..
2024-03-23 23:46:38.201 T:344540   debug <general>: WINDOW MANAGER: LAUNCH ALL SPORTS PAGE
2024-03-23 23:46:38.201 T:344540   debug <general>:

2024-03-23 23:46:38.201 T:344540   debug <general>: WINDOW MANAGER: CLOSE PAGE
2024-03-23 23:46:38.201 T:344540   debug <general>:

2024-03-23 23:46:38.211 T:405     debug <general>: Activating window ID: 13032
2024-03-23 23:46:38.211 T:405     debug <general>: ------ Window Deinit (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/landingpag>
2024-03-23 23:46:38.211 T:405     debug <general>: ------ Window Init (/home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.xm>
2024-03-23 23:46:38.211 T:405      info <general>: Loading skin file: /home/alarm/.kodi/addons/plugin.sportsview/resources/skins/default/1080i/allsports.xml>
2024-03-23 23:46:38.212 T:344540   debug <general>: AllSportsWindow - onInit
2024-03-23 23:46:38.212 T:344540   debug <general>:

2024-03-23 23:46:38.990 T:344540   debug <general>: ALLSPORTS_LIST.TXT AND SPORTS_LIST MATCH
2024-03-23 23:46:38.990 T:344540   debug <general>:

2024-03-23 23:46:38.995 T:344540   debug <general>: ffmpeg[0x7ffeec2e2c98]: [swscaler] No accelerated colorspace conversion found from yuv420p to bgra.
`

**Comment by ZappBrannigannnn** [comment]
Was this a direct copy and paste?

There seem to be no errors but (possibly unimportant because no errors)
there seem to be some characters missing in filenames like allsports.xml is
only "allsports.xm" and later just "allsports." but that would throw an
error for sure.

Maybe try deleting the sportsview folder in temp and try again?

**Comment by ghost** [comment]
I just copied the relevant parts of the log file. I don't see any errors.

I deleted the temp folder and tried again. I can't figure out why it doesn't work. In All Sports if I hit the arrow key I see the American Football button without any text on it. For some reason it doesn't loop through the rest of them. I click on the AFB button and it takes me to a multi-color background page with nothing on it.

**Comment by ZappBrannigannnn** [comment]
That's really interesting and is starting to make me believe it is a Linux
issue or pi limitation.

If I had a Linux machine and knew anything about it I would love to test it
and make it work. Even when I built it on Windows then tested it on my
shield it did require some small changes to make it compatible on both. But
even then it threw errors for me to work with.

I wonder if the pi is struggling? It is a ton of stuff all pulled from the
api website. Does it lock up?

As a side note. The addon in its current state is really only good for the
"my matches" section which I do use for my personal use and may make a fork
to focus and streamline just that. If that's something you want to use it
for we could focus on getting that to work for you?

**Comment by ghost** [comment]
I have a Rpi 5 so probably has enough power. The 3+ I had I would have said possible it wasn't enough power.

We can work on my matches. You could setup a VirtualBox on your windows system and install arch or any other linux distro and then install Kodi and then your app.

**Comment by ZappBrannigannnn** [comment]
When you select "my matches" on the landing page what happens?

Are there any errors at that point?

In configuration what is your sports folder and what are its contents?

**Comment by ghost** [comment]
I am pointing to your demo sports folder.

This does give me an error. 

` Error Type: <class 'TypeError'>
                                                   Error Contents: cannot unpack non-iterable float object
                                                   Traceback (most recent call last):
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 80, in get_sports_folders
                                                       self.check_button_cache()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 103, in check_button_cache
                                                       self.generate_custom_buttons(sport)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 128, in generate_custom_buttons
                                                       text_width, text_height = focused_draw.textlength(sport, font)
                                                       ^^^^^^^^^^^^^^^^^^^^^^^
                                                   TypeError: cannot unpack non-iterable float object
                                                   -->End of Python script error report<--
`

**Comment by ZappBrannigannnn** [comment]
Isn't this from the change you made to the code the other day?

My line 128 looks like this:

            text_width, text_height = focused_draw.textsize(sport, font)

**Comment by ZappBrannigannnn** [comment]
try it again with textsize instead of textlength then post the error for me.

**Comment by ghost** [comment]
` - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'ImageDraw' object has no attribute 'textsize'
                                                   Traceback (most recent call last):
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 80, in get_sports_>
                                                       self.check_button_cache()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 103, in check_butt>
                                                       self.generate_custom_buttons(sport)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 128, in generate_c>
                                                       text_width, text_height = focused_draw.textsize(sport, font)
                                                                                 ^^^^^^^^^^^^^^^^^^^^^
                                                   AttributeError: 'ImageDraw' object has no attribute 'textsize'
                                                   -->End of Python script error report<--
`

**Comment by ZappBrannigannnn** [comment]
it's so strange that on your machine it doesn't like "textsize"... I realize it's deprecated but it still works everywhere else i test. 

Either way, when i change it to "textlength" in my tests it does work and I'll keep it as that since it appears to be the up-to-date term.

Could this have something to do with the version of PIL installed? I was under the impression that the addon.xml file line here:         "import addon="script.module.pil" version="1.1.7"/" set this. But maybe this is just the script that kodi uses to interpret/relay the PIL commands.???

**Comment by ZappBrannigannnn** [comment]
i only changed it in the mysportsbuttons.py line 128 so we may run into this again later.

**Comment by ghost** [comment]
- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'TypeError'>
                                                   Error Contents: cannot unpack non-iterable float object
                                                   Traceback (most recent call last):
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 80, in get_sports_f>
                                                       self.check_button_cache()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 103, in check_butto>
                                                       self.generate_custom_buttons(sport)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 128, in generate_cu>
                                                       text_width, text_height = focused_draw.textlength(sport, font)
                                                       ^^^^^^^^^^^^^^^^^^^^^^^
                                                   TypeError: cannot unpack non-iterable float object
                                                   -->End of Python script error report<--

Same error.

**Comment by ZappBrannigannnn** [comment]
Yeah. I didn't change anything. Only updated it to reflect the change you
had made.

What version of pillow do you have installed?

**Comment by ZappBrannigannnn** [comment]
What is directly in your sports folder? If there are no sports that match
sports in thesportdb api then it'll be empty, therefore no words, therefore
no textlength.

Can you take a screenshot of your sports folder in a file browser and then
a screenshot of your configuration showing the sports folder you've pointed
to in the sportsview configuration?

**Comment by ghost** [comment]
I am pointing to this folder on my RPI

https://github.com/ZappBrannigannnn/SportsView/tree/main/plugin.sportsview/SPORTS

It says my script.module.pil is 5.1 and I am running Kodi 21 RC1

**Comment by ZappBrannigannnn** [comment]
Yes. That's the folder that it should be pointed to. I was hoping to see
what yours had. Earlier I think there was mention of /sports/mlb/etc. this
would not work... It should be sports/baseball/mlb/etc. ... If I could see
a pic of your folder structure to make sure it is good that would eliminate
possible causes. Then we can move to the next step. Also the configuration
screen pic.. It doesn't have to be a screenshot. Just a photo.

**Comment by ZappBrannigannnn** [comment]
The latest version of pillow is 10.2. Is this what you have? Where are you
getting script.module.pil version number?

What is RC1? Like a nightly build or something? Kodi is at 20.5 afaik.
Probably. Not the cause. Could you try installing the current version of
kodi to see if it works?

**Comment by ghost** [comment]
When I right click on the SportsView addon and then info and then dependencies, it display what is needed and what I have. It says I have PIL 5.1. RC stands for release candidate as I am using Kodi 21. I probably can't downgrade. 
![Screenshot_20240326_173949](https://github.com/ZappBrannigannnn/SportsView/assets/65799766/44476d61-232e-4efc-aa61-5c397d2ddf88)

**Comment by ghost** [comment]
The actual python-pillow version I have is 10.2.

**Comment by ZappBrannigannnn** [comment]
okay. this all looks good... let me experiment a bit.

**Comment by ZappBrannigannnn** [comment]
the strangest thing is that when I:

1. delete my folder : "\Kodi\temp\sportsview\my_sports_buttons_cache"
2. change line 128 to "textlength" and go to the "my matches" page 
3. it doesn't work. i get the error:

2024-03-27 12:50:02.704 T:28592   error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'ImageDraw' object has no attribute 'textlength'
                                                   Traceback (most recent call last):
                                                     File "C:\Users\mjc_c\AppData\Roaming\Kodi\addons\plugin.sportsview\mysports\mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "C:\Users\mjc_c\AppData\Roaming\Kodi\addons\plugin.sportsview\mysports\mysportsbuttons.py", line 80, in get_sports_folders
                                                       self.check_button_cache()
                                                     File "C:\Users\mjc_c\AppData\Roaming\Kodi\addons\plugin.sportsview\mysports\mysportsbuttons.py", line 103, in check_button_cache
                                                       self.generate_custom_buttons(sport)
                                                     File "C:\Users\mjc_c\AppData\Roaming\Kodi\addons\plugin.sportsview\mysports\mysportsbuttons.py", line 128, in generate_custom_buttons
                                                       text_width, text_height = focused_draw.textlength(sport, font)
                                                   AttributeError: 'ImageDraw' object has no attribute 'textlength'
                                                   -->End of Python script error report<--

But, when i 

1. delete my folder : "\Kodi\temp\sportsview\my_sports_buttons_cache"
2. change line 128 to "textsize" and go to the "my matches" page
3. It works. 

I thought it might be because my version of PIL was 10.0... I updated it to 10.2 and went through these steps again both ways (textsize and textlength) and still get the same results.

Strange that textlength will not work for me but textsize won't work for you.

**Comment by ZappBrannigannnn** [comment]
This is a strange question.  Can you copy and paste your addon.xml file contents here please?

**Comment by ZappBrannigannnn** [comment]
Also, when you startup kodi does the log say "FindAddons: script.module.pil v5.1.0 installed" ?

**Comment by ghost** [comment]
FindAddons: script.module.pil v5.1.0 installed

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<addon id="plugin.sportsview" name="Sportsview" version="0.6.5+matrix.1" provider-name="ZappBrannigan">

    <requires>
        <import addon="xbmc.python" version="3.0.0"/>
        <import addon="script.module.requests" version="2.25.1"/>
        <import addon="script.module.dateutil" version="2.8.2"/>
        <import addon="script.module.pil" version="1.1.7"/>
        <import addon="script.module.dateutil" version="2.8.2"/>
    </requires>

    <extension point="xbmc.python.script" library="default.py">
        <provides>executable</provides>
    </extension>

    <extension point="xbmc.addon.metadata">
        <summary lang="en_GB">A Kodi add-on for displaying sports information.</summary>
        <description lang="en_GB">This add-on provides information about various sports including leagues, teams, and players.</description>
        <disclaimer lang="en_GB"></disclaimer>
        <language></language>
        <platform>all</platform>
        <license>GPLv2</license>
        <forum>http://forum.example.com</forum>
        <website>http://www.example.com</website>
        <email>contact@example.com</email>
        <source>https://github.com/your-username/your-repository</source>
        <news>http://www.example.com/news</news>
        <assets>
            <icon>icon.png</icon>
            <fanart>fanart.jpg</fanart>
            <banner></banner>
            <clearlogo></clearlogo>
            <screenshot></screenshot>
        </assets>
    </extension>

</addon>
`

**Comment by ZappBrannigannnn** [comment]
Aahhhh... This is driving me crazy!

I asked for help on the kodi forum...

This should not be happening! All the dependencies and versions are the
same for everything. The folders are set properly. For some reason yours is
being strict on textlength and mine is strict on textsize. Neither of which
should really be. Even if it is deprecated all it should show is a warning
that it's deprecated.

**Comment by ZappBrannigannnn** [comment]
Sorry... First time turning on my computer for a while. 

I got the advice to remove the PIL version number from the addon.xml

Try updating, then try again...

**Comment by ghost** [comment]
With textsize changed to textlength

"/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 80, in get_sports_f>
                                                       self.check_button_cache()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 103, in check_butto>
                                                       self.generate_custom_buttons(sport)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 128, in generate_cu>
                                                       text_width, text_height = focused_draw.textlength(sport, font)
                                                       ^^^^^^^^^^^^^^^^^^^^^^^
                                                   TypeError: cannot unpack non-iterable float object
                                                   -->End of Python script error report<--

Textsize:

- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'ImageDraw' object has no attribute 'textsize'
                                                   Traceback (most recent call last):
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportswindow.py", line 23, in onInit
                                                       self.sports_buttons.get_sports_folders()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 80, in get_sports_f>
                                                       self.check_button_cache()
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 103, in check_butto>
                                                       self.generate_custom_buttons(sport)
                                                     File "/home/alarm/.kodi/addons/plugin.sportsview/mysports/mysportsbuttons.py", line 128, in generate_cu>
                                                       text_width, text_height = focused_draw.textsize(sport, font)
                                                                                 ^^^^^^^^^^^^^^^^^^^^^
                                                   AttributeError: 'ImageDraw' object has no attribute 'textsize'
                                                   -->End of Python script error report<--

**Comment by ZappBrannigannnn** [comment]
Looks like the errors didn't change. So the suggestion didn't work.

I'll keep working on it.
